lua/lspconfig/server_configurations/leanls.lua

```diff
@@ -20,6 +24,19 @@ return {
         or util.find_git_ancestor(fname)
     end,
     on_new_config = function(config, root_dir)
+      local lake_version = ''
+      local lake_job = vim.fn.jobstart({ 'lake', '--version' }, {
+        on_stdout = function(_, d, _)
+          lake_version = table.concat(d, '\n')
+        end,
+        stdout_buffered = true,
+      })
+      if lake_job > 0 and vim.fn.jobwait({ lake_job })[1] == 0 then
+        local major = lake_version:match 'Lake version (%d).'
+        if major and tonumber(major) < 3 then
+          config.cmd = legacy_cmd
```

Comment:
I wonder if we should make this overridable (e.g. by adding `legacy_cmd` as a config field)?

Reply:
It's technically already overridable by end-users who want to define `on_new_config` themselves -- but generally to make it even more easily overridable, maybe someday we want `timeout` and `memory` options like VSCode does to assemble the command line, and then at that point could add one for this too -- but I'm not sure I see why it's super useful personally yet otherwise. The vast majority of people will never even know this exists once Lean 4 has another release, no?

Reply:
The `legacy_cmd` is still used when editing files in core lean (which doesn't use lake).  Having a field for extra command-line arguments seems like a good idea (I don't think we need `timeout` and `memory` specifically, writing `-M4000` is just as easy).